apps/emqx_auth_jwt/src/emqx_authn_jwt_schema.erl

```diff
@@ -172,12 +188,21 @@ verify_claims(required) ->
 verify_claims(_) ->
     undefined.
 
-do_check_verify_claims([]) ->
+convert_validated_value(<<"${clientid}">>) -> clientid;
+convert_validated_value(<<"${username}">>) -> username;
```

Comment:
can continue to support binary() placeholders?

And call `emqx_template:lookup_var/2` to resolve value when check.